### ISSUE TITLE
<HeatDirectory  Transforms="Filter.xslt"> doesn't seem to work

### ISSUE THREAD

**Issue by virzak** [wixtoolset/issues#7112]
The .wixproj file contains:

```xml
<HeatDirectory OutputFile="MyApp.Wpf.Binaries.wxs" Directory="$(MyAppPublish)" DirectoryRefId="VIEWER" ComponentGroupName="MyApp.WpfHarvestBinaries" SuppressCom="true" SuppressFragments="true" SuppressRegistry="true" SuppressRootDirectory="true" AutoGenerateGuids="false" GenerateGuidsNow="true" ToolPath="$(WixToolPath)" PreprocessorVariable="var.MyApp.Wpf.TargetDir" Transforms="Filter.xslt" />
```

And `Filter.xslt` contains:
```xml
  <xsl:key name="service-search" match="wix:Component[wix:File/@Source = '$(var.MyApp.Wpf.TargetDir)\MyApp.exe']" use="@Id" />
```

Wix3 skips generation of MyApp.exe as it should and everything builds fine.
Wix4, however, generates Component and File nodes for MyApp.exe, which results in error:

```
Q:\Code\Novitech\MyApp\Wix\MyApp.Installer\Components.wxs(8): error WIX0204: ICE30: The target file 'MyApp.exe' is installed in '[ProgramFilesFolder]\Zomp\MyApp\Viewer\' by two different components on an SFN system: 'cmpIkkBUuBJIDAZWDKQYJthWuney4A' and 'MyAppExe'. This breaks component reference counting. 
```

**Comment by robmen** [comment]
We don't use GitHub Issues for support questions. [Please pick from one of the support options presented here](https://www.firegiant.com/support-options/).

It is possible through discussion that a specific bug in the WiX Toolset will be uncovered but usually, someone can help you find the error in your .wxs code.

**Comment by virzak** [comment]
Turns out that it was a namespace `xmlns="http://schemas.microsoft.com/wix/2006/wi"` in the xslt file. The `wix convert` didn't change it to a new namespace.

**Comment by robmen** [comment]
`wix convert` doesn't convert project files, only source files. [HeatWave][hw] does convert projects and [this issue](https://github.com/firegiant/HeatWaveSupport/issues/5) was fixed in the latest version 0.9.1

[hw]: https://www.firegiant.com/wix/heatwave/
